subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -137,4 +140,18 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
             return DefaultBuildTreeWorkGraphPreparer(buildRegistry, buildTaskSelector)
         }
     }
+
+    private
+    class TaskExecutionAccessCheckerProvider(
+        private val isConfigurationCacheEnabled: Boolean
+    ) {
+        fun createTaskExecutionAccessChecker(
+            configurationTimeBarrier: ConfigurationTimeBarrier,
+            configurationCacheStartParameter: List<ConfigurationCacheStartParameter>
```

Comment:
❓ Why a list of start parameters?

Reply:
If it's a parameter of `ConfigurationCacheStartParameter`, then it fails to satisfy the requirements when the service is not registered. The `List<...>` is used to ignore the absence of `ConfigurationCacheStartParameter` in non-CC builds.

Reply:
I see. Clever.

Reply:
I made it more clear in https://github.com/gradle/gradle/pull/23210/commits/7a7d164ec05ff0df0e79a79334f9fdfb322f65f5 anyway.